halo2/src/prover.rs

```diff
@@ -312,10 +311,6 @@ fn gen_proof<
     circuit: C,
     instances: &[Vec<Fr>],
 ) -> Vec<u8> {
-    MockProver::run(params.k(), &circuit, instances.to_vec().clone())
-        .unwrap()
-        .assert_satisfied();
```

Comment:
This doesn't really do anything except print nice error messages if the constraints are not satisfied. I think that would be nice to have, but then we can't test the proving in isolation. Maybe it's fair that if you want to debug failing constraints you have to use the mock prover, similar to how you'd use Pilcom instead of eStark on the Goldilocks field?

Reply:
yea makes sense